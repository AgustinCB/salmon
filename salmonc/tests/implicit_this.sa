class Test {
  var internal = "initial";
  testProp1() {
    fun accessor() {
      print this.internal;
    }
    return accessor;
  }
  getter testProp() {
    return internal;
  }
  setter testProp(a) {
    internal = a;
    return a;
  }
}

var obj = Test();
print obj.testProp;
obj.testProp = "testProp 111";
print obj.testProp;
obj.testProp1()();
