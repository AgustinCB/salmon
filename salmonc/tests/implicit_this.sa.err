Constants: [SimpleValue(Function { ip: 107, arity: 0, uplifts: None }), SimpleValue(Integer(1)), SimpleValue(Integer(4)), SimpleValue(Integer(3)), SimpleValue(String(0)), SimpleValue(String(1)), SimpleValue(String(9)), SimpleValue(Nil), SimpleValue(Function { ip: 132, arity: 1, uplifts: None }), SimpleValue(Function { ip: 141, arity: 1, uplifts: None }), SimpleValue(Function { ip: 154, arity: 2, uplifts: None }), SimpleValue(String(25)), SimpleValue(String(41)), SimpleValue(String(51)), SimpleValue(String(60)), SimpleValue(String(70)), SimpleValue(String(78)), SimpleValue(String(94)), SimpleValue(String(104)), SimpleValue(Object { address: 120, tags: 280 }), SimpleValue(String(287)), SimpleValue(String(294)), SimpleValue(String(300)), SimpleValue(String(307)), SimpleValue(String(319)), SimpleValue(String(336))]
Instructions: [Instruction { instruction_type: Constant(0), location: 1 }, Instruction { instruction_type: SetGlobal(1), location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(8), location: 3 }, Instruction { instruction_type: SetGlobal(2), location: 3 }, Instruction { instruction_type: Pop, location: 3 }, Instruction { instruction_type: Constant(9), location: 4 }, Instruction { instruction_type: SetGlobal(3), location: 4 }, Instruction { instruction_type: Pop, location: 4 }, Instruction { instruction_type: Constant(10), location: 6 }, Instruction { instruction_type: SetGlobal(4), location: 6 }, Instruction { instruction_type: Pop, location: 6 }, Instruction { instruction_type: Constant(20), location: 6 }, Instruction { instruction_type: Constant(21), location: 6 }, Instruction { instruction_type: Constant(19), location: 6 }, Instruction { instruction_type: AddTag, location: 6 }, Instruction { instruction_type: AddTag, location: 6 }, Instruction { instruction_type: Constant(22), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: Constant(5), location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: ObjectSet, location: 7 }, Instruction { instruction_type: Swap, location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: SetGlobal(5), location: 7 }, Instruction { instruction_type: Pop, location: 7 }, Instruction { instruction_type: GetGlobal(5), location: 8 }, Instruction { instruction_type: Call, location: 8 }, Instruction { instruction_type: SetGlobal(0), location: 8 }, Instruction { instruction_type: Pop, location: 8 }, Instruction { instruction_type: Constant(4), location: 9 }, Instruction { instruction_type: Constant(16), location: 9 }, Instruction { instruction_type: GetGlobal(0), location: 9 }, Instruction { instruction_type: ObjectHas, location: 9 }, Instruction { instruction_type: JmpIfFalse(5), location: 9 }, Instruction { instruction_type: Constant(16), location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: ObjectGet, location: 9 }, Instruction { instruction_type: Call, location: 9 }, Instruction { instruction_type: Jmp(3), location: 9 }, Instruction { instruction_type: Constant(15), location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: ObjectGet, location: 9 }, Instruction { instruction_type: ToStr, location: 9 }, Instruction { instruction_type: StringConcat, location: 9 }, Instruction { instruction_type: Push, location: 9 }, Instruction { instruction_type: Strlen, location: 9 }, Instruction { instruction_type: Swap, location: 9 }, Instruction { instruction_type: Constant(1), location: 9 }, Instruction { instruction_type: Constant(3), location: 9 }, Instruction { instruction_type: Constant(2), location: 9 }, Instruction { instruction_type: Syscall, location: 9 }, Instruction { instruction_type: Pop, location: 9 }, Instruction { instruction_type: Constant(23), location: 10 }, Instruction { instruction_type: Constant(18), location: 10 }, Instruction { instruction_type: GetGlobal(0), location: 10 }, Instruction { instruction_type: ObjectHas, location: 10 }, Instruction { instruction_type: JmpIfFalse(5), location: 10 }, Instruction { instruction_type: Constant(18), location: 10 }, Instruction { instruction_type: Swap, location: 10 }, Instruction { instruction_type: ObjectGet, location: 10 }, Instruction { instruction_type: Call, location: 10 }, Instruction { instruction_type: Jmp(4), location: 10 }, Instruction { instruction_type: Constant(15), location: 10 }, Instruction { instruction_type: Swap, location: 10 }, Instruction { instruction_type: ObjectSet, location: 10 }, Instruction { instruction_type: Pop, location: 10 }, Instruction { instruction_type: Pop, location: 10 }, Instruction { instruction_type: Constant(4), location: 11 }, Instruction { instruction_type: Constant(16), location: 11 }, Instruction { instruction_type: GetGlobal(0), location: 11 }, Instruction { instruction_type: ObjectHas, location: 11 }, Instruction { instruction_type: JmpIfFalse(5), location: 11 }, Instruction { instruction_type: Constant(16), location: 11 }, Instruction { instruction_type: Swap, location: 11 }, Instruction { instruction_type: ObjectGet, location: 11 }, Instruction { instruction_type: Call, location: 11 }, Instruction { instruction_type: Jmp(3), location: 11 }, Instruction { instruction_type: Constant(15), location: 11 }, Instruction { instruction_type: Swap, location: 11 }, Instruction { instruction_type: ObjectGet, location: 11 }, Instruction { instruction_type: ToStr, location: 11 }, Instruction { instruction_type: StringConcat, location: 11 }, Instruction { instruction_type: Push, location: 11 }, Instruction { instruction_type: Strlen, location: 11 }, Instruction { instruction_type: Swap, location: 11 }, Instruction { instruction_type: Constant(1), location: 11 }, Instruction { instruction_type: Constant(3), location: 11 }, Instruction { instruction_type: Constant(2), location: 11 }, Instruction { instruction_type: Syscall, location: 11 }, Instruction { instruction_type: Pop, location: 11 }, Instruction { instruction_type: Constant(24), location: 12 }, Instruction { instruction_type: GetGlobal(0), location: 12 }, Instruction { instruction_type: ObjectHas, location: 12 }, Instruction { instruction_type: JmpIfFalse(5), location: 12 }, Instruction { instruction_type: Constant(24), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectGet, location: 12 }, Instruction { instruction_type: Call, location: 12 }, Instruction { instruction_type: Jmp(3), location: 12 }, Instruction { instruction_type: Constant(13), location: 12 }, Instruction { instruction_type: Swap, location: 12 }, Instruction { instruction_type: ObjectGet, location: 12 }, Instruction { instruction_type: Call, location: 12 }, Instruction { instruction_type: Call, location: 12 }, Instruction { instruction_type: Pop, location: 12 }, Instruction { instruction_type: Return, location: 12 }, Instruction { instruction_type: Constant(4), location: 1 }, Instruction { instruction_type: Constant(6), location: 1 }, Instruction { instruction_type: GetLocal(0), location: 1 }, Instruction { instruction_type: ObjectHas, location: 1 }, Instruction { instruction_type: JmpIfFalse(5), location: 1 }, Instruction { instruction_type: Constant(6), location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: ObjectGet, location: 1 }, Instruction { instruction_type: Call, location: 1 }, Instruction { instruction_type: Jmp(3), location: 1 }, Instruction { instruction_type: Constant(5), location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: ObjectGet, location: 1 }, Instruction { instruction_type: ToStr, location: 1 }, Instruction { instruction_type: StringConcat, location: 1 }, Instruction { instruction_type: Push, location: 1 }, Instruction { instruction_type: Strlen, location: 1 }, Instruction { instruction_type: Swap, location: 1 }, Instruction { instruction_type: Constant(1), location: 1 }, Instruction { instruction_type: Constant(3), location: 1 }, Instruction { instruction_type: Constant(2), location: 1 }, Instruction { instruction_type: Syscall, location: 1 }, Instruction { instruction_type: Pop, location: 1 }, Instruction { instruction_type: Constant(7), location: 1 }, Instruction { instruction_type: Return, location: 1 }, Instruction { instruction_type: Uplift(0), location: 2 }, Instruction { instruction_type: Constant(1), location: 2 }, Instruction { instruction_type: ArrayAlloc, location: 2 }, Instruction { instruction_type: MultiArraySet, location: 2 }, Instruction { instruction_type: AttachArray(1), location: 2 }, Instruction { instruction_type: Constant(7), location: 2 }, Instruction { instruction_type: Pop, location: 2 }, Instruction { instruction_type: GetGlobal(1), location: 3 }, Instruction { instruction_type: Return, location: 3 }, Instruction { instruction_type: Constant(6), location: 4 }, Instruction { instruction_type: GetLocal(0), location: 4 }, Instruction { instruction_type: ObjectHas, location: 4 }, Instruction { instruction_type: JmpIfFalse(5), location: 4 }, Instruction { instruction_type: Constant(6), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Call, location: 4 }, Instruction { instruction_type: Jmp(3), location: 4 }, Instruction { instruction_type: Constant(5), location: 4 }, Instruction { instruction_type: Swap, location: 4 }, Instruction { instruction_type: ObjectGet, location: 4 }, Instruction { instruction_type: Return, location: 4 }, Instruction { instruction_type: GetLocal(1), location: 5 }, Instruction { instruction_type: Constant(11), location: 5 }, Instruction { instruction_type: GetLocal(0), location: 5 }, Instruction { instruction_type: ObjectHas, location: 5 }, Instruction { instruction_type: JmpIfFalse(5), location: 5 }, Instruction { instruction_type: Constant(11), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectGet, location: 5 }, Instruction { instruction_type: Call, location: 5 }, Instruction { instruction_type: Jmp(4), location: 5 }, Instruction { instruction_type: Constant(5), location: 5 }, Instruction { instruction_type: Swap, location: 5 }, Instruction { instruction_type: ObjectSet, location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: Pop, location: 5 }, Instruction { instruction_type: GetLocal(1), location: 6 }, Instruction { instruction_type: Return, location: 6 }]
Locations: [Location { address: 336, line: 3 }, Location { address: 336, line: 4 }, Location { address: 336, line: 0 }, Location { address: 336, line: 6 }, Location { address: 336, line: 9 }, Location { address: 336, line: 12 }, Location { address: 336, line: 13 }, Location { address: 336, line: 1 }, Location { address: 336, line: 17 }, Location { address: 336, line: 18 }, Location { address: 336, line: 19 }, Location { address: 336, line: 20 }, Location { address: 336, line: 21 }]
