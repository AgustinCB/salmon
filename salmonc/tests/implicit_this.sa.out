initial
testProp 111
testProp 111
